src/cbmc_starter_kit/setup_ci.py

```diff
@@ -0,0 +1,163 @@
+        'help': "Use this tag of cadical in CI. default: %(default)s"
+        }]
+    args = arguments.create_parser(
+        options=options,
+        description=desc,
+        formatter_class=RawDescriptionHelpFormatter).parse_args()
+    arguments.configure_logging(args)
+    return args
+
+################################################################
+
+def _read_file_template(path):
+    """Read file and return list of lines"""
+    with open(path, encoding='utf-8') as data:
+        return data.read().splitlines()
+
+def _write_file_template(lines, path):
+    """Write file at specified path with the provided list of lines"""
+    with open(path, "w", encoding='utf-8') as data:
+        data.write('\n'.join(lines) + '\n')
+
+def _replace_placeholders_in_workflow_template(lines, replacements):
+    """Returns a list of new lines, where some lines have had a placeholder
+    value appropriately updated."""
+    buf = []
+    cmpl = re.compile(r"\s+(?P<key>[\w_]+):\s<__[\w_]+__>")
+    for line in lines:
+        if not line or line[len(line) - 1] != ">":
+            # Append immediately if empty line or line does not end with '>'
+            buf.append(line)
+            continue
+        if line.strip() == "runs-on: <__GITHUB_ACTIONS_RUNNER__>":
+            key = "GITHUB_ACTIONS_RUNNER"
+            buf.append(line.replace(f"<__{key}__>", replacements[key]))
+            continue
+        match = cmpl.match(line)
+        if match is None:
+            buf.append(line)
+        else:
+            key = match['key']
+            buf.append(line.replace(f"<__{key}__>", replacements[key]))
+    return buf
+
+def patch_proof_ci_workflow(args, proof_ci_path):
+    """Patch GitHub Actions workflow with appropriate values"""
+    lines = _read_file_template(proof_ci_path)
+    proofs_dir = repository.get_relative_path_from_repository_to_proofs_root()
+    replacements = {
+        "CADICAL_TAG": args.cadical,
+        "CBMC_VERSION": args.cbmc,
+        "CBMC_VIEWER_VERSION": args.cbmc_viewer,
+        "GITHUB_ACTIONS_RUNNER": args.github_actions_runner,
+        "KISSAT_TAG": args.kissat,
+        "LITANI_VERSION": args.litani,
+        "PROOFS_DIR": proofs_dir,
+    }
+    new_lines = _replace_placeholders_in_workflow_template(lines, replacements)
+    _write_file_template(new_lines, proof_ci_path)
+
+################################################################
+
+def add_or_update_summarize_module(repo_root, quiet=False):
+    """Will either add or update the "summarize" module located within the CBMC
+    proof root's lib module"""
+    logging.debug('Updating CBMC starter kit')
+    src = util.package_repository_template_root() / util.PROOF_DIR / util.LIB_MODULE
+    dst = repository.get_abspath_to_proofs_root(repo_root) / util.LIB_MODULE
+    (logging.debug if quiet else logging.info)('Updating summarize script: %s -> %s', src, dst)
+    assert src.exists()
+    shutil.copytree(src, dst, dirs_exist_ok=True)
```

Comment:
> This isn't just updating the summarize module, it's copying an entire directory called lib. Why isn't this method called that?

It can be easily renamed. `lib` currently maintains only `summarize`, but you're right, it ought to be renamed. I will rename.

> Also, if we're expecting other things to go in lib, then shouldn't this method be in util? 

I don't expect other modules to be added to `lib`. I suggest moving the method to `util` if it indeed gets used in more than one place.

> And I don't understand "add_or_update", this is just going to clobber every existing file in the lib directory and add all new ones, right?

The function will either add every file in `lib` or will update any file of the same that is already present in the destination.